make getToken public to simplify interacting with the API

anarcat opened the pull request:
this is necessary to POST new entries for example

Strubbl commented on token.go:
```diff
@@ -20,6 +20,8 @@ type Token struct {
 }
 
+// GetToken will use the credentials set in the configuration to
+// request an acces token from the Wallabag API
```
please change: 
* acces --> access
* Wallabag --> wallabag

Strubbl reviewed (commented):
Sorry, i just saw your PR now. I was working in parallel. Can you please rebase?

anarcat committed:
make getAuthTokenHeader() API public for external use

anarcat commented:
okay - i completely rewrote this since you introduced a much more useful function that I can use directly. in the end, maybe all this will be made useless by #5, but for now it would be nice to see this merged anyways...

anarcat committed:
add missing docs

we also include getToken docs since a previous version of this patched GetToken instead

Strubbl commented:
OKay, let's merge that for now. Later on a refactoring to your suggested doAPI() (in #5) will be done. Thanks!
